### DISCUSSION TITLE
What't the different of     SURFACE and the POST_PROCESS?

### DISCUSSION THREAD

**Discussion by dugufly** [google/filament#4619] [Q&A]
What't the different of     SURFACE and the POST_PROCESS?

```
SURFACE         = 0, //!< shaders applied to renderables
POST_PROCESS    = 1, //!< shaders applied to rendered buffers
```

**Comment by romainguy** [comment]
Post process are materials specific to post-processing and take a color buffer as an input. Surface materials are materials applied to regular meshes renderer in the scene.

**Reply by dugufly** [reply]
Are there relevant documents? I don't quite understand

**Reply by romainguy** [reply]
This is internal details of Filament at this point so there are no relevant documents. Post processing is used to implement things like blur, bloom, color grading, etc.

**Reply by dugufly** [reply]
If I want to render to the specified FBO (not on the screen), is there an interface that can be implemented?

**Reply by romainguy** [reply]
To render to an off screen texture, you can use Filament's `RenderTarget` API. See our [render target sample](https://github.com/google/filament/blob/main/samples/rendertarget.cpp#L213). After rendering in the `RenderTarget` you can use the `Texture` as an input for a material (again, see the sample).

**Reply by dugufly** [reply]
If the gltf model is used, and the MaterialProvider uses UbershaderLoader (only this mode is available in Android), the default material ubershader.mat.in will be used. 
1. Can this default material(ubershader.mat.in) be changed to other materials? 

2. In this example (gltf_viewer. cpp), the defaultMaterial will be loaded . What is the function of this material？
     `    RenderableManager::Builder(1)
            .material(0, shadowMaterial->getDefaultInstance())`

**Reply by romainguy** [reply]
> Can this default material(ubershader.mat.in) be changed to other materials?

After loading the glTF file you can replace the materials on all renderables. You can also provide your own `MaterialProvider` and use/create your own materials however you want.

> In this example (gltf_viewer. cpp), the defaultmaterial will be loaded . What is the function of this material？

What material are you talking about?

**Reply by dugufly** [reply]
I mean this ：[https://github.com/google/filament/blob/main/samples/gltf_viewer.cpp#L322]

Why are there multiple materials？ ubershader.mat.in & defaultMaterial.mat

**Reply by romainguy** [reply]
That's not defaultMaterial.mat it's the default instance of a ground shadow material. It's an option of the viewer to show a shadow on a virtual ground.

**Reply by dugufly** [reply]
1. Load the gltf model,  Even if I turn off the lighting in the gltf, It still run the LIT model. How can I change it to the UNLIT model?
2. How to change base color texture during rendering when I use gltf model?
